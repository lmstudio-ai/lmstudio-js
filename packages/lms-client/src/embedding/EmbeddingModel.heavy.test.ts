import { LMStudioClient, type EmbeddingModel } from "../index.js";
import { embeddingTestingNomic, ensureHeavyTestsEnvironment } from "../shared.heavy.test.js";

describe("EmbeddingModel", () => {
  let client: LMStudioClient;
  let model: EmbeddingModel;
  beforeAll(async () => {
    client = new LMStudioClient();
    await ensureHeavyTestsEnvironment(client);
  });
  beforeEach(async () => {
    model = await client.embedding.model(embeddingTestingNomic, { verbose: false });
  });
  it("can embed a single string", async () => {
    const result = await model.embed("Chaos is a ladder.");
    expect(result.embedding.length).toMatchInlineSnapshot(`768`);
    expect(result.embedding[0]).toBeCloseTo(0.037533789, 4);
  });
  it("can embed multiple strings", async () => {
    const result = await model.embed([
      "Chaos is a ladder.",
      "Two chaoses are two ladders.",
      "Three chaoses are three ladders.",
    ]);
    expect(result.length).toBe(3);
    expect(result[0].embedding.length).toMatchInlineSnapshot(`768`);
    expect(result[0].embedding[0]).toBeCloseTo(0.037533789, 4);
    expect(result[1].embedding.length).toMatchInlineSnapshot(`768`);
    expect(result[1].embedding[0]).toBeCloseTo(0.007641574, 4);
    expect(result[2].embedding.length).toMatchInlineSnapshot(`768`);
    expect(result[2].embedding[0]).toBeCloseTo(0.008860555, 4);
  });
  it("can get model context length", async () => {
    const contextLength = await model.getContextLength();
    expect(contextLength).toMatchInlineSnapshot(`2048`);
  });
  it("can get model info", async () => {
    const modelInfo = await model.getModelInfo();
    expect(modelInfo).toMatchSnapshot({
      identifier: expect.any(String),
      instanceReference: expect.any(String),
      modelKey: expect.any(String),
    });
  });
  it("Can tokenize correctly", async () => {
    const tokens = await model.tokenize("Chaos is a ladder.");
    expect(tokens).toMatchSnapshot();
  });
  it("Can tokenize multiple strings correctly", async () => {
    const tokens = await model.tokenize([
      "Cersei understands the consequences of her absence",
      "and she is absent anyway",
    ]);
    expect(tokens).toMatchSnapshot();
  });
  it("Can count tokens correctly", async () => {
    const tokenCount = await model.countTokens("Chaos is a ladder.");
    expect(tokenCount).toMatchInlineSnapshot(`5`);
  });
});
