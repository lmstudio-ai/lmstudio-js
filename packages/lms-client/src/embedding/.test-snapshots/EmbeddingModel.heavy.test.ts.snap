// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmbeddingModel Can tokenize correctly 1`] = `
[
  101,
  8488,
  2003,
  1037,
  10535,
  1012,
  102,
]
`;

exports[`EmbeddingModel Can tokenize multiple strings correctly 1`] = `
[
  [
    101,
    8292,
    22573,
    2072,
    19821,
    1996,
    8465,
    1997,
    2014,
    6438,
    102,
  ],
  [
    101,
    1998,
    2016,
    2003,
    9962,
    4312,
    102,
  ],
]
`;

exports[`EmbeddingModel can get model info 1`] = `
{
  "architecture": "nomic-bert",
  "contextLength": 2048,
  "displayName": "Nomic Embed Text v1.5",
  "format": "gguf",
  "identifier": Any<String>,
  "instanceReference": Any<String>,
  "maxContextLength": 2048,
  "modelKey": Any<String>,
  "path": "nomic-ai/nomic-embed-text-v1.5-GGUF/nomic-embed-text-v1.5.Q4_K_M.gguf",
  "sizeBytes": 84106624,
  "type": "embedding",
}
`;
